core/services/relay/evm/method_binding.go

```diff
@@ -28,9 +31,22 @@ func (m *methodBinding) SetCodec(codec commontypes.RemoteCodec) {
 	m.codec = codec
 }
 
-func (m *methodBinding) Bind(_ context.Context, binding commontypes.BoundContract) error {
-	m.address = common.HexToAddress(binding.Address)
+func (m *methodBinding) Bind(ctx context.Context, binding commontypes.BoundContract) error {
+	addr := common.HexToAddress(binding.Address)
+
+	// check for contract byte code at the latest block and provided address
+	byteCode, err := m.client.CodeAt(ctx, addr, nil)
+	if err != nil {
+		return err
+	}
+
+	if len(byteCode) == 0 {
+		return fmt.Errorf("%w: %s", ErrNoContractExists, addr)
```

Comment:
How about a custom error?
```suggestion
		return ErrNoContractExists{Address: addr}
```
```go
type ErrNoContractExists {
	Address common.Address
}

func (e ErrNoContractExists) Error() string {
	return fmt.Sprintf("contract does not exist at address: %s", addr)
}
```